cpp/src/arrow/python/builtin_convert.cc

```diff
@@ -157,8 +182,14 @@ class ScalarVisitor {
   int64_t float_count_;
   int64_t binary_count_;
   int64_t unicode_count_;
+  int64_t decimal_count_;
+
+  int32_t max_decimal_precision_;
+  int32_t max_decimal_scale_;
+
   // Place to accumulate errors
   // std::vector<Status> errors_;
+  OwnedRefNoGIL decimal_type_;
```

Comment:
This member doesn't seem used actually. Were you planning to use it with `PyObject_IsInstance` instead of the costlier call to `internal::PyDecimal_Check`?

Reply:
Yes, thanks. Will fix.